lib/direct_message_channel_creator.rb

```diff
@@ -40,4 +45,64 @@ def self.update_memberships(unique_target_users, chat_channel_id)
       membership.save!
     end
   end
+
+  def self.ensure_actor_can_communicate!(acting_user, target_users)
+    # We never want to prevent the actor from communicating with themself.
+    target_users = target_users.reject { |user| user.id == acting_user.id }
+
+    UserCommScreener
+      .new(acting_user: acting_user, target_user_ids: target_users.map(&:id))
+      .preventing_actor_communication
+      .each do |user_id|
+        raise NotAllowed.new(
+                I18n.t(
+                  "chat.errors.not_accepting_dms",
+                  username: target_users.find { |user| user.id == user_id }.username,
+                ),
+              )
+      end
+
+    # TODO (martin) Add methods from here and below to UserCommScreener in core to
```

Comment:
This can be done in a separate polish PR, don't want to hold this up because core needs another API on the class I added.